src/types/from_register_input.rs

```diff
@@ -1,13 +1,94 @@
 use std::time::Duration;
 
+use bytes::{BufMut, Bytes, BytesMut};
 use compositor_pipeline::{
-    pipeline::{self, decoder, input},
+    pipeline::{
+        self, decoder,
+        input::{self, rtp::InputAudioStream},
+    },
     queue,
 };
 
 use super::register_input::*;
 use super::util::*;
 
+/// [RFC 3640, section 4.1. MIME Type Registration (`config` subsection)](https://datatracker.ietf.org/doc/html/rfc3640#section-4.1)
+fn parse_asc(asc: Option<&str>) -> Result<Option<Bytes>, TypeError> {
+    let Some(asc) = asc else {
+        return Ok(None);
+    };
+
+    const EMPTY_ASC: &str = "The AudioSpecificConfig field is empty. Either omit this field or provide a non-empty ASC.";
+    if asc.is_empty() {
+        return Err(TypeError::new(EMPTY_ASC));
+    }
+
+    let mut output = BytesMut::new();
+
+    const BAD_ASC_FORMAT: &str = "Not all of the provided AudioSpecificConfig is hex digits.";
+    if !asc.chars().all(|c| char::is_ascii_hexdigit(&c)) {
+        return Err(TypeError::new(BAD_ASC_FORMAT));
+    }
+
+    for a in asc.as_bytes().chunks(2) {
+        let fallback = &[a[0], 0];
+        let val = if a.len() == 2 { a } else { fallback };
+        // the value is already checked for being ascii
+        let val = String::from_utf8_lossy(val);
```

Comment:
```suggestion
        let val = match a {
           [b1, b2, ..] => &[b1, b2],
           [b1] => &[b1, 0],
           [] => &[0, 0],
        };
        // the value is already checked for being ascii
        let val = String::from_utf8_lossy(val);
```

Btw is it even correct that there would be an odd number of hex digits?

Reply:
> Btw is it even correct that there would be an odd number of hex digits?
Sort of. The spec says that a non-byte aligned asc should be padded with zeroes. I'm not sure whether the receiver or the sender should add the padding, but it doesn't hurt to make sure on our side, especially since this is a well defined behavior